opendatahub-io/data-science-pipelines-operator
# Add ResourceRequirments to WorkflowController

- Introduces ResourceRequirements for WorkflowController in DSPA CRD
- Adds dynamic request and limit defaults for WC deployment

## The issue resolved by this Pull Request:
Resolves [RHOAIENG-8242](https://issues.redhat.com/browse/RHOAIENG-8242)

## Description of your changes:



## Testing instructions
- Deploy DSPO
- Create DSPA, preferably with custom values set for `spec.workflowController.resources.requests.memory`, `spec.workflowController.resources.limits.memory`, `spec.workflowController.resources.requests.cpu`, and `spec.workflowController.resources.limits.cpu`
- Go to OpenShift Console, go to DSPA project, and click Pods
- Find pod with name starting with `ds-pipeline-workflow-controller-` and click it
- Go to Metrics tab, ensure the Memory Usage and CPU Usage graphs show correct limits
- Run several simultaneous pipelines, ensure the limits are not encroached upon

## Checklist
